how-to/how-to-use-virtio-mem-with-kata.md

```diff
@@ -0,0 +1,49 @@
+# Kata Containers with virtio-mem
+
+- [Introduction](#introduction)
+- [Pre-requisites](#pre-requisites)
+- [Run a Kata Container with utilizing virtio-mem](#run-a-kata-container-utilizing-virtio-mem)
+
+## Introduction
+
+The basic idea of virtio-mem is to provide a flexible, cross-architecture memory hot(un)plug solution that avoids many limitations imposed by existing technologies, architectures, and interfaces. More details can be found in https://lkml.org/lkml/2019/12/12/681.
+
+Kata Containers with virtio-mem support memory resize.
+
+## Pre-requisites
+
+Kata Containers with virtio-mem requires Linux and the qemu that support virtio-mem.
+
+The Linux kernel is at https://github.com/davidhildenbrand/linux/tree/virtio-mem-rfc-v4.
+The Linux kernel config that can work with Kata Containers is at https://gist.github.com/teawater/016194ee84748c768745a163d08b0fb9.
+
+The qemu is at https://github.com/teawater/qemu/tree/kata-virtio-mem. (The original source is at https://github.com/davidhildenbrand/qemu/tree/virtio-mem.  Its base version of qemu cannot work with Kata Containers.  So merge the commit of virtio-mem to upstream qemu.)
+
+Set Linux and the qemu that support virtio-mem with following line in the Kata Containers configuration.
+```
+path = "qemu-dir"
+kernel = "vmlinux-dir"
+```
+
+Enable virtio-mem with following line in the Kata Containers configuration.
+```
+enable_virtio_mem = true
+```
+
+## Run a Kata Container utilizing virtio-mem
+
+Use following command to enable memory overcommitment of a Linux kernel.  Because qemu virtio-mem device need to allocate a lot of memory.
+```
+echo 1 > /proc/sys/vm/overcommit_memory
```

Comment:
Once we have resizable memory backends and can resize memory blocks while the guest is running (I have patches for both already that I am getting onto shape right now), that step will be optional.

Reply:
Great!  Please let me know when patches can be test.  I'd love to try them.

Reply:
Could you use change this to the use `sudo` - something like the following:

```sh
$ echo 1 | sudo tee /proc/sys/vm/overcommit_memory
```

Reply:
done.